tests/Integration/Behaviour/Features/Scenario/Product/Shop/bulk_delete_product_multishop.feature

```diff
@@ -44,9 +44,9 @@ Feature: Copy product from shop to shop.
       | delta_quantity | 51 |
     And I add new image "image1" named "app_icon.png" to product "standardProduct" for shop "shop2"
     And I add new image "image2" named "some_image.jpg" to product "standardProduct" for shop "shop2"
-    And I copy product standardProduct from shop shop2 to shop shop1
-    And I copy product standardProduct from shop shop2 to shop shop3
-    And I copy product standardProduct from shop shop2 to shop shop4
+    When I set following shops for product "standardProduct":
+      | source shop | shop2             |
+      | shops       | shop1,shop3,shop4 |
```

Comment:
```suggestion
      | shops       | shop1,shop2,shop3,shop4 |
```
Shouldn't the source always be in the list? This actually comforts me in the comment I left on the command constructor, that we should assert source shop ID is part of the associated shop IDs list.

Reply:
https://github.com/PrestaShop/PrestaShop/pull/31512#discussion_r1121457916
answered there (we can continue discussion in that comment if you still not convinced)

Reply:
Regarding your answer:
> but source shop must be checked against initial shop ids taken from db, and the source shop will be skipped anyway if it exists in selectedShops array, so I don't see a point forcing to add the shop into "selectedShops" when its always filtered out in handler anyway.

The idea is to keep some consistency because of the name of the command `SetProductShopsCommand`:
- source shop ID is the source used to copy the data
- shop IDs represent all the associated shops of the product, it can even contain some shop already associated if we don't want to remove their association So the source ID is necessarily one of them, which is why I find it weird if it's not in the list that defines all the associations

I know it would work even if it's not inside based on the implementation, but it's just to make the contract a bit clearer. I'm already not 100% sure of the naming because of this source shop ID, but I don't have a better one, but at least the content of the parameters should be very clear and explicit.

Reply:
Besides in all the other behat tests (unless I missed some) you already followed this convention to have the source inside the list, which makes sense, the purpose is to force this convention to avoid any confusion

In the same way, you can't define a default category that isn't in the list of associated categories (we can always force considering it's inside, but it seems even clearer to force it - at least in a CQRS command)